fuzon-http/src/main.rs

```diff
@@ -75,28 +75,34 @@ async fn top(data: web::Data<AppState>, req: web::Query<CodeRequest>) -> Result<
     Ok(web::Json(top_terms))
 }
 
+/// http server to serve the fuzon terminology matching api
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    /// Path to the configuration file.
+    #[clap(short, long)]
+    config: String,
+}
+
+
 #[actix_web::main] // or #[tokio::main]
 async fn main() -> std::io::Result<()> {
+    let args = Args::parse();
+    let config_path = args.config;
+
+    let config: Config = serde_json::from_reader(
+        File::open(config_path).expect("Failed to open config file.")
+    ).expect("Failed to parse config.");
 
-    // NOTE: config as inline json for debugging, later this will be a config file
-    let config_data = r#"
-        {
-            "collections": {
-                "cell_type": "https://purl.obolibrary.org/obo/cl.owl",
-                "source_material": "https://purl.obolibrary.org/obo/uberon.owl",
-                "taxon_id": "https://purl.obolibrary.org/obo/ncbitaxon/subsets/taxslim.owl"
-            }
-        }"#;
     let data = web::block(move || 
-        serde_json::from_str::<Config>(config_data).unwrap()
+        AppState::from_config(config)
```

Comment:
**note(non-blocking)**: I would need an explanation (in person) on why this solution suddently fixes the concurrency problems. Good job finding it out ;)

Reply:
I think it works because of `web::block`. I was not able to use it before, because it requires cloning the instance of `AppState`, along with its internal `TermMatcher`s and they did not implement `Clone`, after deriving Clone on everything, this solution worked :p